### DISCUSSION TITLE
How to modify MB1 & MB2 BCTs and how to sign and store with keys

### DISCUSSION THREAD

**Discussion by kunoh** [OE4T/meta-tegra#1565] [Q&A]
Hi,

I'm trying to figure out how to modify the BCTs or any other modifiable parts of the Jetson Orin Nano boot process when using meta-tegra.

Is there a guide I'm missing that explain how to do it and also explain how to use keys to sign these parts?

**Comment by ichergui** [comment]
Hi @kunoh 
You shouldn't modify MB1 and MB2. 
Could you please elaborate about the changes that you need ? 
Are you referring to secure boot ?  if so, following our wiki page below 
https://github.com/OE4T/meta-tegra/wiki/Secure-Boot-Support-in-L4T-R35.2.1-and-later

**Comment by kunoh** [comment]
I may missunderstand the documentation, but for example the last line under MB1 in https://docs.nvidia.com/jetson/archives/r35.4.1/DeveloperGuide/text/AR/BootArchitecture/JetsonOrinSeriesBootFlow.html#mb1 says the following; 

> NVIDIA owns MB1, and provides it as a binary in the Jetson BSP package. You can configure its behavior for a specific platform through its Boot Configuration Table, MB1-BCT.

Can I configure the MB1-BCT through meta-tegra ?
And similar for MB2-BCT there is also some configurables for the pinmux for example, that may need changes when building a custom board?

I don't have any specific examples at the moment, I'm just trying to figure out how it is all coupled together :)

**Reply by tschoop** [reply]
Yes. There are several possibilities to control/configure the behavior of MB1 and MB2 even though the respective software are closed binaries. For example, Nvidia offers a spreadsheet to configure the pin functionalities (pinmux, gpio configuration, ..) which contain macros to generate dtsi files. These DTS formatted files are configuration files for MB1. Other files exist for MB2 control, for example, the access to the eeprom. If you already have a build running you can have a look into tegraflash archive. This contains these configuration files for the different platforms. They are generated out of meta-tegra BSP layer. If you go the yocto way you can create your own layer to patch/substitute/modify these configurations.
